mmv1/third_party/terraform/tests/resource_assured_workloads_workload_test.go.erb

```diff
@@ -0,0 +1,168 @@
+<% autogen_exception -%>
+package google
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	dcl "github.com/GoogleCloudPlatform/declarative-resource-client-library/dcl"
+<% unless version == 'ga' -%>
+	assuredworkloads "github.com/GoogleCloudPlatform/declarative-resource-client-library/services/google/assuredworkloads/beta"
+<% else -%>
+	assuredworkloads "github.com/GoogleCloudPlatform/declarative-resource-client-library/services/google/assuredworkloads"
+<% end -%>
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccAssuredWorkloadsWorkload_basic(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"random_suffix":   randString(t, 10),
+		"org_id":          getTestOrgFromEnv(t),
+		"billing_account": getTestBillingAccountFromEnv(t),
+	}
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAssuredWorkloadsWorkloadDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAssuredWorkloadsWorkload_basic(context),
+			},
+			{
+				ResourceName:            "google_assured_workloads_workload.meep",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"billing_account"},
+			},
+			{
+				Config: testAccAssuredWorkloadsWorkload_basicUpdate(context),
+			},
+			{
+				ResourceName:            "google_assured_workloads_workload.meep",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"billing_account"},
+			},
+		},
+	})
+}
+
+func TestAccAssuredWorkloadsWorkload_full(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"random_suffix":   randString(t, 10),
+		"org_id":          getTestOrgFromEnv(t),
+		"billing_account": getTestBillingAccountFromEnv(t),
+	}
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAssuredWorkloadsWorkloadDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAssuredWorkloadsWorkload_full(context),
+			},
+			{
+				ResourceName:            "google_assured_workloads_workload.meep",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"billing_account", "kms_settings", "resource_settings", "provisioned_resources_parent"},
+			},
+		},
+	})
+}
+
+func testAccAssuredWorkloadsWorkload_basic(context map[string]interface{}) string {
+	return Nprintf(`
+resource "google_assured_workloads_workload" "meep" {
+	provider = google-beta
+	display_name = "workloadExample"
+	billing_account = "billingAccounts/%{billing_account}"
+	compliance_regime = "FEDRAMP_MODERATE"
```

Comment:
Can we test a different regime between basic/full?

Reply:
nope we can't test a difference compliance regime at the moment. It required us to be enrolled upstream which is in progress but will take a week or so.